src/Microsoft.ML.Data/Transforms/Hashing.cs

```diff
@@ -1388,7 +1388,7 @@ private bool SaveAsOnnxCore(OnnxContext ctx, int iinfo, string srcVariable, stri
                 // Since these numeric types are not supported by Onnxruntime, we cast them to UInt32.
                 if (srcType == NumberDataViewType.UInt16 || srcType == NumberDataViewType.Int16 ||
                     srcType == NumberDataViewType.SByte || srcType == NumberDataViewType.Byte ||
-                    srcType == BooleanDataViewType.Instance)
+                    srcType == BooleanDataViewType.Instance || srcType is KeyDataViewType)
```

Comment:
>srcType is KeyDataViewType [](start = 63, length = 26)

For the key type case, you also need to check that `RawType` is either `byte` or `ushort`, because for `uint` and `ulong` you don't need to add the cast node.

Reply:
Sorry, I was looking at the previous iteration.

---
In reply to: [430171432](https://github.com/dotnet/machinelearning/pull/5160#discussion_r430171432) [](ancestors = 430171432)